src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.Internal.VisualStudio.Shell.ErrorList;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.FindUsages
+{
+    internal static class RichNavOptions
+    {
+        internal const string RichNavAvailableOptionName = "RichNavAvailable";
+    }
+
+    [Export(typeof(IScopeFilterFactory))]
+    [TableManagerIdentifier("FindAllReferences*")]
+    [TableManagerIdentifier("FindResults*")]
+    [Replaces(PredefinedScopeFilterNames.EntireSolutionScopeFilter)]
+    [Replaces(PredefinedScopeFilterNames.AllItemsScopeFilter)]
+    [DeferCreation(OptionName = RichNavOptions.RichNavAvailableOptionName)] // This factory will not be loaded unless this option is set to Boolean true
```

Comment:
does c++ (or other non-roslyn language) have richnav, if so, could this get loaded in c++ scenarios?  (same q for the rest below)

Reply:
good question.  @AmadeusW ?

Reply:
@AmadeusW ?

Reply:
it won't be loaded because of the `DeferCreation` Attribute. We fetch `OptionName` from the MEF Metadata and see whether this option is defined. RichNav is responsible for setting this option, and they won't set it in C++ scenarios

Reply:
@BertanAygun corrected me: in 16.7 preview 1, RichNav will set this option even in C++. I recommended that we don't set that option.

Long term, when Roslyn FAR moves to LSP, we will have a mechanism for LSP to hook up to "user selected broader scope" event and restart the search with new parameters